prime-router/src/main/kotlin/fhirengine/engine/FHIRReceiver.kt

```diff
@@ -0,0 +1,358 @@
+                        sender.topic,
+                        sender.schemaName
+                    )
+                )
+            )
+        }
+    }
+
+    private fun validateSubmissionMessage(
+        sender: Sender,
+        actionLogger: ActionLogger,
+        queueMessage: FhirReceiveQueueMessage,
+    ): Report? {
+        val rawReport = BlobAccess.downloadContent(queueMessage.blobURL, queueMessage.digest)
+        return if (rawReport.isBlank()) {
+            actionLogger.error(InvalidReportMessage("Provided raw data is empty."))
+            null
+        } else {
+            val report: Report
+            val sources = listOf(ClientSource(organization = sender.organizationName, client = sender.name))
+
+            when (sender.format) {
+                MimeFormat.HL7 -> {
+                    val messages: List<Message> = HL7Reader(actionLogger).getMessages(rawReport)
+                    val isBatch: Boolean = HL7Reader(actionLogger).isBatch(rawReport, messages.size)
+                    // create a Report for this incoming HL7 message to use for tracking in the database
+
+                    report = Report(
+                        if (isBatch) MimeFormat.HL7_BATCH else MimeFormat.HL7,
+                        sources,
+                        messages.size,
+                        metadata = metadata,
+                        nextAction = TaskAction.convert,
+                        topic = sender.topic,
+                        id = queueMessage.reportId,
+                        bodyURL = queueMessage.blobURL
+                    )
+
+                    // TODO fix and re-enable https://github.com/CDCgov/prime-reportstream/issues/14103
+                    // dupe detection if needed, and if we have not already produced an error
+    //                if (!allowDuplicates && !actionLogs.hasErrors()) {
+    //                    doDuplicateDetection(
+    //                        workflowEngine,
+    //                        report,
+    //                        actionLogs
+    //                    )
+    //                }
+
+                    // check for valid message type
+                    messages.forEachIndexed { idx, element ->
+                        MessageType.validateMessageType(element, actionLogger, idx + 1)
+                    }
+                }
+
+                MimeFormat.FHIR -> {
+                    val bundles = FhirTranscoder.getBundles(rawReport, actionLogger)
+                    report = Report(
+                        MimeFormat.FHIR,
+                        sources,
+                        bundles.size,
+                        metadata = metadata,
+                        nextAction = TaskAction.convert,
+                        topic = sender.topic,
+                        id = queueMessage.reportId,
+                        bodyURL = queueMessage.blobURL
+                    )
+                }
+
+                MimeFormat.CSV -> {
```

Comment:
This should probably be handled by the else below and change text to: `Unsupported` instead of `Unexpected`?